tool/tbot/main.go

```diff
@@ -227,11 +228,29 @@ func onStart(botConfig *config.BotConfig) error {
 	reloadChan := make(chan struct{})
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-
 	go handleSignals(log, reloadChan, cancel)
 
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if err := sendTelemetry(
+			ctx, telemetryClient(os.Getenv), os.Getenv, log, botConfig,
+		); err != nil {
+			log.WithError(err).Error(
+				"Failed to send anonymous telemetry.",
+			)
+		}
+	}()
+
 	b := tbot.New(botConfig, log, reloadChan)
-	return trace.Wrap(b.Run(ctx))
+	if err := trace.Wrap(b.Run(ctx)); err != nil {
+		return err
+	}
+
+	// Ensures telemetry finishes sending before application exits in oneshot.
+	wg.Wait()
```

Comment:
Is it intended that we don't wait for the `sendTelemetry` goroutine to end if `b.Run()` returns an error?

Conversely, should we add a timeout after the end of a successful `Run` to exit even if `sendTelemetry` isn't done?

Reply:
I wasn't too sure if we'd want to hold up a failed invocation in order to submit telemetry as I could see that being frustrating. I like your idea of adding a timeout, and, as this uses the context cancelled by signals they should be able to CTRL-C out of it.

Reply:
```
INFO [TBOT]      Waiting up to 30s for anonymous telemetry to finish sending before exiting. Press CTRL-C to cancel. tbot/main.go:253
^CINFO [TBOT]      Received interrupt, canceling... tbot/main.go:280
WARN [TBOT]      Anonymous telemetry transmission cancelled due to signal or timeout. tbot/main.go:261
```

Reply:
4533ff5d09